src/providers/exploreState.tsx

```diff
@@ -494,14 +502,11 @@ function exploreReducer(
       const filterCount = getFilterCount(filterState);
       return {
         ...state,
-        entityPageState: {
-          ...state.entityPageState,
-          [state.tabValue]: {
-            ...state.entityPageState[state.tabValue],
-            filterCount,
-            filterState,
-          },
-        },
+        entityPageState: updateEntityPageState(
+          state.tabValue,
+          state.entityPageState,
+          { filterCount, filterState }
```

Comment:
`updateEntityPageState` function to update the entity's `entityPageState` with the updated `filterCount` and `filterState`.